src/state/axisSelectors.ts

```diff
@@ -94,27 +176,192 @@ export const selectAxisDomain = (
     return selectNumericalDomain(state, axisSettings, axisType, axisId);
   }
 
-  const allDataSquished = selectAllDataSquished(state, axisType, axisId, axisSettings.dataKey);
+  const allDataSquished = selectAllDataSquished(state, axisType, axisId);
   return computeCategoricalDomain(allDataSquished, axisSettings);
 };
 
+/**
```

Comment:
man where did all this stuff exist before 😅

Reply:
Yeah it's wild. What I discovered so far is:

- scale gets created in `generateCategoricalChart` first .. somewhere
- then `formatAxisMap` takes it and changes a little
- then the ticks generator also takes the scale and changes its domain based on the ticks it generated

I think it's a prime example of why we should wrap the d3 scales in something immutable 😬

Reply:
yeah...